pkg/controllers/pod_controller.go

```diff
@@ -180,6 +185,9 @@ func (r *PodReconciler) handleRestartPolicy(ctx context.Context, pod corev1.Pod,
 		if err := r.Get(ctx, types.NamespacedName{Name: leaderPodName, Namespace: pod.Namespace}, &leader); err != nil {
 			return false, err
 		}
+		if leader.Labels[leaderworkerset.TemplateRevisionHashKey] != pod.Labels[leaderworkerset.TemplateRevisionHashKey] {
```

Comment:
Add a comment explaining why we do this: basically if the revision is different from the leader, then we expect this pod will be deleted soon and an alternative will be created using the new revision.